docs/madr/decisions/054-meshservice-identity-sync.md

```diff
@@ -0,0 +1,75 @@
+# Sync MeshService identity for cross-zone communication
+
+* Status: accepted
+
+## Context and Problem Statement
+
+In order to establish mTLS communication a source service needs to verify SAN of the destination service.
+Currently, we rely on autogenerated `kuma.io/service` tag, which long term will be replaced.
+We encode all Kuma tags of a Dataplane proxy object in the certificate that we generate for a proxy.
+Our initial plan was to require all tags selected by MeshService. For example
+
+```yaml
+spec:
+  selector:
+    dataplaneTags: # tags in Dataplane object, see below
+      app: redis
+      version: v1
+```
+
+We would require cert with tags `app=redis` and `version=v1`.
+However, there is no construct in Envoy that would verify SAN of `kuma://app/redis` **and** `kuma://version/v1`.
+We can only verify SAN 1 or SAN 2.
+Therefore, we need to "fallback" to `kuma.io/service`. 
+
+## Considered Options
+
+* Update MeshService with identity
+  * Store in `status`
+  * Store in `spec`
+
+## Decision Outcome
+
+Update MeshService with identity and store it in `spec`. 
+
+## Pros and Cons of the Options
+
+### Update MeshService with identity and sync it
+
+We need to have a place in MeshService to set an identity. We can specify a new field.
+
+```yaml
+spec: # or status, see discussion below
+  identities:
+  - type: ServiceTag
+    value: redis_kuma-demo_svc_80
+```
+
+Currently, the only type is `ServiceTag` but we want this API to be extensible to future changes of identity.
+
+This is a list because one MeshService can select multiple Dataplane proxies of multiple different `kuma.io/service`, so for mTLS we need to require either one of them.
+
+We need a component that updates this value in `MeshService` object.
+The component would go over all Dataplane objects, match proxies for MeshService and add all `kuma.io/service` tags to this list.
+Additionally unrelated to this MADR, we have to introduce a component that is equivalent of `ServiceInsight` data (count of offline, online dpps etc.) in Service status.
+This data also requires all matched Dataplane objects therefore it can be the same component.
```

Comment:
Can we mention the future where these might just be correct spiffe ids?

Reply:
added